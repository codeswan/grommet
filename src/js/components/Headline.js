// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React, { PropTypes, Component } from 'react';
import classnames from 'classnames';
import CSSClassnames from '../utils/CSSClassnames';

const CLASS_ROOT = CSSClassnames.HEADLINE;

export default class Headline extends Component {
  render () {
    let classes = classnames(
      CLASS_ROOT,
      {
        [`${CLASS_ROOT}--${this.props.size}`]: this.props.size,
        [`${CLASS_ROOT}--align-${this.props.align}`]: this.props.align,
        [`${CLASS_ROOT}--margin-${this.props.margin}`]: this.props.margin,
        [`${CLASS_ROOT}--strong`]: this.props.strong
      },
      this.props.className
    );

    return (
      <div className={classes}>
        {this.props.children}
      </div>
    );
  }
};

Headline.propTypes = {
  align: PropTypes.oneOf(['start', 'center', 'end']),
  margin: PropTypes.oneOf(['none', 'small', 'medium', 'large']),
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  strong: PropTypes.bool
};
