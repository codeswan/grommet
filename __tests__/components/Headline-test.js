// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import Headline from '../../src/js/components/Headline';

describe('Headline', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Headline>Testing</Headline>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
