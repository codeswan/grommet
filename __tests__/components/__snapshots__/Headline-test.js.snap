exports[`Headline has correct default options 1`] = `
<div
  className="grommetux-headline">
  Testing
</div>
`;
